arrow-core-data/src/main/kotlin/arrow/typeclasses/Foldable.kt

```diff
@@ -211,16 +211,21 @@ interface Foldable<F> {
    * Get the first element of the foldable or none if empty or the predicate does not match
    */
   fun <A> Kind<F, A>.firstOption(predicate: (A) -> Boolean): Option<A> =
-    get(0).filter(predicate)
+    foldLeft<A, Either<A, Long>>(0L.right()) { i, a ->
+      i.flatMap {
+        if (predicate(a)) Left(a)
+        else Right(it + 1L)
+      }
+    }.swap().toOption()
```

Comment:
While this works it's not great for very large structures, especially not lazy ones like sequence because `foldLeft` won't stop after it found something.

Can you redo this with `foldRight`? It should be possible to just write something like:
```kotlin
foldRight(Eval.now(None)) { x, evalAcc ->
  if (p(x)) Eval.now(x.some())
  else evalAcc
}
```

Reply:
I see! I'll change it!

Reply:
Although isn't that the exact same impl of `find` ?

Reply:
> Although isn't that the exact same impl of `find` ?

True, did not know that existed. It'll still be good to match the stdlib with `firstOption` I think. So defining it as an alias might not bee too bad. The stdlib has `find` as well which by the looks of it does the same thing as `firstOrNull`. 

I think doing `firstOption(p) == find(p)` will work fine. Oh and please change `firstOption() == find { true }`. The less we use `get` the better. Then we can get rid of the explicit overrides for `firstOption` here: https://github.com/arrow-kt/arrow-core/blob/009099e6c07312b60188dadf7b9ea84d831eea32/arrow-core/src/main/kotlin/arrow/core/extensions/sequenceK.kt#L166-L180

Reply:
@1Jajen1 thanks for the explanation!! :D

Reply:
Thanks for changing this^^ This has been annoying me for a long time :)